src/tests/ftest/datamover/dm_obj_small.py

```diff
@@ -0,0 +1,254 @@
+                    c_value = ctypes.create_string_buffer(data)
+                    c_size = ctypes.c_size_t(ctypes.sizeof(c_value))
+                    ioreq.single_insert(c_dkey, c_akey, c_value, c_size)
+
+                for akey_idx in range(self.num_akeys_array_per_dkey):
+                    # Round-robin to get the size of data and
+                    # the number of extents, and
+                    # arbitrarily use a number 0-9 to fill data
+                    akey_size_idx = akey_idx % len(self.akey_sizes)
+                    data_size = self.akey_sizes[akey_size_idx]
+                    akey_extent_idx = akey_idx % len(self.akey_extents)
+                    num_extents = self.akey_extents[akey_extent_idx]
+                    c_data = []
+                    akey = "akey array {}".format(akey_idx)
+                    c_akey = ctypes.create_string_buffer(akey)
+                    for data_idx in range(num_extents):
+                        data_val = str(data_idx % 10)
+                        data = data_size * data_val
+                        c_data.append([ctypes.create_string_buffer(data), data_size])
+                    ioreq.insert_array(c_dkey, c_akey, c_data)
+            obj.close()
+        cont.close()
+
+    def verify_dataset(self, cont):
+        """Verify the dataset.
+
+        Args:
+            cont (TestContainer): the container
+
+        """
+        self.log.info("Verifying dataset in %s/%s",
+            str(cont.pool.uuid), str(cont.uuid))
+
+        cont.open()
+
+        for obj_idx in range(self.num_objs):
+            obj = DaosObj(cont.pool.context, cont.container, self.obj_list[obj_idx].c_oid)
+            obj.open()
+            ioreq = IORequest(cont.pool.context, cont.container, obj)
+            for dkey_idx in range(self.num_dkeys_per_obj):
+                dkey = "dkey {}".format(dkey_idx)
+                c_dkey = ctypes.create_string_buffer(dkey)
+
+                for akey_idx in range(self.num_akeys_single_per_dkey):
+                    # Round-robin to get the size of data and
+                    # arbitrarily use a number 0-9 to fill data
+                    akey_size_idx = akey_idx % len(self.akey_sizes)
+                    data_size = self.akey_sizes[akey_size_idx]
+                    data_val = str(akey_idx % 10)
+                    data = data_size * data_val
+                    akey = "akey single {}".format(akey_idx)
+                    c_akey = ctypes.create_string_buffer(akey)
+                    buf = ioreq.single_fetch(c_dkey, c_akey, data_size + 1)
+                    actual_data = str(buf.value)
+                    if actual_data != data:
+                        self.log.info("Expected:\n{}\nBut got:\n{}".format(
+                            data[:100] + "...",
+                            actual_data[:100] + "..."))
+                        self.log.info(
+                            "For:\nobj: {}.{}\ndkey: {}\nakey: {}".format(
```

Comment:
(pylint-logging-format-interpolation) Use lazy % formatting in logging functions